### PR TITLE
[Minor] Optimize ShuffleServerInfo#hashCode

### PR DESCRIPTION
**Pull request by xianjingfeng**
### What changes were proposed in this pull request?

Optimize ShuffleServerInfo#hashCode

### Why are the changes needed?

For better performence.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No need

### CODE REVIEW

File: common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public int getPort() {
 
   @Override
   public int hashCode() {
-    return host.hashCode();
+    return id.hashCode();
```
Comment (kaijchen) [inline common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:57]: Why not include all three fields (`id`, `host`, `port`) in hashCode()?
Such as `Objects.hash(id, host, port)`.

Comment (xianjingfeng) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:57]: By default `id = host + "-" + port`, so id is enough.

Comment (kaijchen) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:57]: > By default `id = host + "-" + port`, so id is enough.

Should we remove the constructor at line 37?

Comment (xianjingfeng) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:57]: No. In some cases, the id does not need to be recalculated.

Comment (kaijchen) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:57]: OK. Can you add some comments here to make future reviewer be aware of it.

Comment (xianjingfeng) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:58]: Done

**Review by zuston — APPROVED** [review]

**Review by jerqi — APPROVED** [review]
LGTM

**Review by kaijchen — APPROVED** [review]

**Pull request merged**